libs/langchain/langchain/hub.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Dict, Optional
+
+from langchainhub import Client
+
+from langchain.load.dump import dumps
+from langchain.load.load import loads
+from langchain.utils import get_from_dict_or_env, get_from_env
+
+
+def _get_client(kwargs: Dict[str, Any]) -> Client:
+    api_url = get_from_dict_or_env(kwargs, "api_url", "LANGCHAIN_HUB_API_URL")
+
+    # read hub-specific api key
+    api_key = get_from_dict_or_env(kwargs, "api_key", "LANGCHAIN_HUB_API_KEY", None)
+
+    # use default api key as fallback
+    if api_key is None:
+        api_key = get_from_env("api_key", "LANGCHAIN_API_KEY")
+    return Client(api_url, api_key=api_key)
+
+
+def push(
+    repo_full_name: str,
+    parent_commit_hash: Optional[str],
```

Comment:
is parent commit hash needed?

Reply:
It's a tradeoff between forcing people to be explicit about which changes come after each other and straightforward being able to immediately "push."

The situation having it avoids is 2 people pushing to the same repo and one of them silently overwriting the other one. In this case, whoever pushes second gets an error (at least while we're doing linear versioning).

I'll put up a doc on this. @nfcampos has some thoughts too.